signature.go

```diff
@@ -131,13 +131,10 @@ func ReadSignature(r io.Reader) (*SignatureType, error) {
 		}
 
 		strongSum := make([]byte, strongLen)
-		n, err := r.Read(strongSum)
+		_, err := io.ReadAtLeast(r, strongSum, int(strongLen))
```

Comment:
Maybe this could be `io.ReadFull()`?

Reply:
Hi @matejvasek ! Sorry for the delay to get into this! This is a good PR, thanks!

Yes, I think `io.ReadFull()` would be slightly better here. Are you still willing to update this PR? (Otherwise I guess can cherry-pick your commit and change to `ReadFull()` on top of it).

Reply:
I did update.